include/pmacc/assert.hpp

```diff
@@ -55,4 +59,16 @@
  */
 #    define PMACC_ASSERT_MSG(expr, msg) (!!(expr)) ? ((void) 0) : pmacc::abortWithError(#    expr, __FILE__, __LINE__, msg)
 
+/** assert check
+ *
+ * if `NDEBUG` is not defined: macro expands to (void)0
+ * in kernel asserts are supported only for cuda device compatibility >= 2.
+ * @param expr expression to be evaluated
+ */
+#if((BOOST_LANG_CUDA && PMACC_CUDA_ARCH < 200 ))
+#    define PMACC_DEVICE_ASSERT(expr) ((void) 0)
+#else
+// HIP is supposed to support asserts
```

Comment:
This comment can be removed. Yes HIP is supporting assert on device side, with the side effect that you require more registers but you guarded the assert that we do not need more registers if we compile in release mode.